content/scripts/build.js

```diff
@@ -1163,23 +1263,37 @@ class Builder {
       renderedHtml = rawHtml;
     } else {
       let flaws;
-      [renderedHtml, flaws] = await this.renderMacros(
-        source,
-        rawHtml,
-        metadata
-      );
+      try {
+        [renderedHtml, flaws] = await this.renderMacros(
+          source,
+          rawHtml,
+          metadata
+        );
+      } catch (err) {
+        throw err;
+      }
       if (flaws.length) {
-        if (this.options.flaws === FLAWS_LEVELS.ERROR) {
+        if (this.options.flawLevels.get("macros") === FLAW_LEVELS.ERROR) {
           // Report and exit immediately on the first document with flaws.
           this.logger.error(
-            chalk.red.bold(`\nFlaws within ${mdnUrlLC} while rendering macros:`)
+            chalk.red.bold(
+              `Flaws (${flaws.length}) within ${mdnUrlLC} while rendering macros:`
+            )
           );
-          for (const flaw of flaws) {
+          flaws.forEach((flaw, i) => {
+            this.logger.error(chalk.bold.red(`${i + 1}: ${flaw.name}`));
             this.logger.error(chalk.red(`${flaw}\n`));
-          }
+          });
+          // XXX This is probably the wrong way to bubble up.
           process.exit(1);
-        } else {
-          this.recordFlaws("macros", mdnUrlLC, flaws);
+        } else if (this.options.flawLevels.get("macros") === FLAW_LEVELS.WARN) {
+          // For each flaw, inject the path of the file that was used.
+          // This gets used in the dev UI so that you can get a shortcut
+          // link to open that file directly in your $EDITOR.
+          flaws.forEach((flaw) => {
+            flaw.filepath = rawHtmlFilepath;
```

Comment:
Since I've already changed my live-sample work to add the `filepath` to flaws of prerequisites, I think this should change to:
```suggestion
            if (!flaw.filepath) {
              flaw.filepath = rawHtmlFilepath;
            }
```
so we don't overwrite those filepaths.